internal/util/logging/logram_test.go

```diff
@@ -0,0 +1,178 @@
+					Level:      1,
+					Time:       time.Date(2022, 12, 31, 11, 59, 1, 0, time.UTC),
+					LoggerName: "logger_1",
+					Message:    "message 1",
+				},
+			},
+		}, {
+			inLog: zapcore.Entry{
+				Level:      2,
+				Time:       time.Date(2022, 12, 31, 11, 59, 2, 0, time.UTC),
+				LoggerName: "logger_2",
+				Message:    "message 2",
+			},
+			expected: []zapcore.Entry{
+				{
+					Level:      1,
+					Time:       time.Date(2022, 12, 31, 11, 59, 1, 0, time.UTC),
+					LoggerName: "logger_1",
+					Message:    "message 1",
+				},
+				{
+					Level:      2,
+					Time:       time.Date(2022, 12, 31, 11, 59, 2, 0, time.UTC),
+					LoggerName: "logger_2",
+					Message:    "message 2",
+				},
+			},
+		}, {
+			inLog: zapcore.Entry{
+				Level:      3,
+				Time:       time.Date(2022, 12, 31, 11, 59, 3, 0, time.UTC),
+				LoggerName: "logger_3",
+				Message:    "message 3",
+			},
+			expected: []zapcore.Entry{
+				{
+					Level:      1,
+					Time:       time.Date(2022, 12, 31, 11, 59, 1, 0, time.UTC),
+					LoggerName: "logger_1",
+					Message:    "message 1",
+				},
+				{
+					Level:      2,
+					Time:       time.Date(2022, 12, 31, 11, 59, 2, 0, time.UTC),
+					LoggerName: "logger_2",
+					Message:    "message 2",
+				},
+				{
+					Level:      3,
+					Time:       time.Date(2022, 12, 31, 11, 59, 3, 0, time.UTC),
+					LoggerName: "logger_3",
+					Message:    "message 3",
+				},
+			},
+		}, {
+			inLog: zapcore.Entry{
+				Level:      4,
+				Time:       time.Date(2022, 12, 31, 11, 59, 4, 0, time.UTC),
+				LoggerName: "logger_4",
+				Message:    "message 4",
+			},
+			expected: []zapcore.Entry{
+				{
+					Level:      2,
+					Time:       time.Date(2022, 12, 31, 11, 59, 2, 0, time.UTC),
+					LoggerName: "logger_2",
+					Message:    "message 2",
+				},
+				{
+					Level:      3,
+					Time:       time.Date(2022, 12, 31, 11, 59, 3, 0, time.UTC),
+					LoggerName: "logger_3",
+					Message:    "message 3",
+				},
+				{
+					Level:      4,
+					Time:       time.Date(2022, 12, 31, 11, 59, 4, 0, time.UTC),
+					LoggerName: "logger_4",
+					Message:    "message 4",
+				},
+			},
+		}, {
+			inLog: zapcore.Entry{
+				Level:      5,
+				Time:       time.Date(2022, 12, 31, 11, 59, 5, 0, time.UTC),
+				LoggerName: "logger_5",
+				Message:    "message 5",
+			},
+			expected: []zapcore.Entry{
+				{
+					Level:      3,
+					Time:       time.Date(2022, 12, 31, 11, 59, 3, 0, time.UTC),
+					LoggerName: "logger_3",
+					Message:    "message 3",
+				},
+				{
+					Level:      4,
+					Time:       time.Date(2022, 12, 31, 11, 59, 4, 0, time.UTC),
+					LoggerName: "logger_4",
+					Message:    "message 4",
+				},
+				{
+					Level:      5,
+					Time:       time.Date(2022, 12, 31, 11, 59, 5, 0, time.UTC),
+					LoggerName: "logger_5",
+					Message:    "message 5",
+				},
+			},
+		},
+	} {
+		name := fmt.Sprintf("AppendGet_%d", n)
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			logram.append(&tc.inLog)
```

Comment:
no. internal zap tests use `assert.Equal`

Reply:
Please don't mark conversations as resolved: it makes it hard to read your answer if you hide it.

I'm not sure how the fact that zap tests use `assert.Equal` relates to my question.

What I mean is that instead of calling `logram.append` directly, we should test that the normal code indeed calls this function in the end. We can do that by calling `Setup` and using `zap.Info/Error/etc` to log messages that would end up in the buffer.

Reply:
Given the test below, do we need that one?

Reply:
in the test below, we check the work of the hooks and we are not able to check the complete comparison of the record, only by the message
At the top, we compare the full match of records.

we can extend the tests below and check for additional fields in the record, but ... 
we can't compare the times because we do not know the exact time in the generated message, 
name of the logger and the logging level will be constant.

tests intersect, but not overlap each other completely